support/controlplane-component/defaults.go

```diff
@@ -101,62 +101,101 @@ func (c *controlPlaneWorkload[T]) defaultOptions(cpContext ControlPlaneContext,
 	return deploymentConfig, nil
 }
 
-func (c *controlPlaneWorkload[T]) applyWatchedResourcesAnnotation(cpContext ControlPlaneContext, podTemplate *corev1.PodTemplateSpec) error {
-	if c.rolloutSecretsNames == nil && c.rolloutConfigMapsNames == nil {
-		return nil
-	}
-	// remove duplicate entries if any.
-	secretsNames := sets.New(c.rolloutSecretsNames...)
-	configMapsNames := sets.New(c.rolloutConfigMapsNames...)
-
-	var hashedData []string
-	for secretName := range secretsNames {
-		secret := &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      secretName,
-				Namespace: cpContext.HCP.Namespace,
-			},
-		}
-		if err := cpContext.Client.Get(cpContext, client.ObjectKeyFromObject(secret), secret); err != nil {
-			if apierrors.IsNotFound(err) {
-				continue
+func podConfigMapNames(spec *corev1.PodSpec) []string {
+	names := sets.New[string]()
+	for _, v := range spec.Volumes {
+		switch {
+		case v.ConfigMap != nil:
+			names.Insert(v.ConfigMap.Name)
+		case v.Projected != nil:
+			for _, source := range v.Projected.Sources {
+				if source.ConfigMap != nil {
+					names.Insert(source.ConfigMap.Name)
+				}
 			}
-			return err
-		}
-
-		for _, value := range secret.Data {
-			hashedData = append(hashedData, util.HashSimple(value))
 		}
 	}
+	return sets.List(names)
+}
 
-	for configMapName := range configMapsNames {
-		configMap := &corev1.ConfigMap{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      configMapName,
-				Namespace: cpContext.HCP.Namespace,
-			},
-		}
-		if err := cpContext.Client.Get(cpContext, client.ObjectKeyFromObject(configMap), configMap); err != nil {
-			if apierrors.IsNotFound(err) {
-				continue
+func podSecretNames(spec *corev1.PodSpec) []string {
```

Comment:
should we pass a client to this function and make it just return a list of secrets directly, instead of having to create a generic fetchResource.

Reply:
The reason I split them is that it should be easier to test whether we're returning the right resources given a podSpec without involving a client. The intent was more/easier ut, which reminds me that I didn't include UTs for this, will add.

Reply:
ack